swift/WorkflowTesting/Sources/WorkflowRenderTester.swift

```diff
@@ -288,6 +291,18 @@
             }
         }
 
+        func runSideEffect(key: AnyHashable, action: (Lifetime) -> Void) {
+            guard let sideEffectIndex = expectations.expectedSideEffects.firstIndex(where: { (expectedSideEffect) -> Bool in
```

Comment:
Should expected side effects be a dictionary instead?

Reply:
Yes, that'd be more correct. However, the API felt weird.

Reply:
Which part felt weird? Creating the expected side effects in your tests? (If so, I think it's fine to keep those as an array and just change the storage. :shrug:)

Reply:
Gotcha! Yes, that makes sense, I'll make that change.

Reply:
Done.